src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -1460,30 +1570,78 @@ void doDocument(DocumentManagement documentManagement,
             new Object[] { opentextDocId, fileDataSize });
         return;
     }
-    long versionNumber = version.getNumber();
-    String contextId = documentManagement.getVersionContentsContext(
-        documentNode.getID(), versionNumber);
-    ContentService contentService =
-        this.soapFactory.newContentService(documentManagement);
-    DataHandler dataHandler = contentService.downloadContent(contextId);
-
     String contentType = version.getMimeType();
     if (contentType != null) {
       response.setContentType(contentType);
     }
-    InputStream inputStream = dataHandler.getInputStream();
-    try {
-      if (inputStream != null) {
-        IOHelper.copyStream(inputStream, response.getOutputStream());
+
+    switch (this.downloadMethod) {
+      case "contentserver": {
+        String urlString = this.indexingContentServerUrl
+            + "?func=ll&objAction=download&objId=" + documentNode.getID();
+        log.log(Level.FINEST, "Downloading content from {0}",
+            urlString);
+        sendUrlResponse(response, urlString,
+            this.soapFactory.getAuthenticationToken(documentManagement));
+        break;
       }
-    } finally {
-      if (inputStream != null) {
+
+      case "cws": {
+        log.log(Level.FINEST, "Downloading content from CWS");
+        long versionNumber = version.getNumber();
+        String contextId = documentManagement.getVersionContentsContext(
+            documentNode.getID(), versionNumber);
+        ContentService contentService =
+            this.soapFactory.newContentService(documentManagement);
+        DataHandler dataHandler = contentService.downloadContent(contextId);
+        InputStream inputStream = dataHandler.getInputStream();
         try {
-          inputStream.close();
-        } catch (IOException ioException) {
-          log.log(Level.FINE, "Error closing document stream", ioException);
+          if (inputStream != null) {
+            IOHelper.copyStream(inputStream, response.getOutputStream());
+          }
+        } finally {
+          if (inputStream != null) {
+            try {
+            inputStream.close();
+          } catch (IOException ioException) {
+            log.log(Level.FINE, "Error closing document stream", ioException);
+          }
         }
       }
+      break;
+     }
+
+     case "cwscontenthandler": {
+      long versionNumber = version.getNumber();
+      String contextId = documentManagement.getVersionContentsContext(
+          documentNode.getID(), versionNumber);
+      String urlString = this.downloadHandlerUrl + "?token="
+          +  escapeParam(this.soapFactory.getAuthenticationToken(
+                  documentManagement))
+          + "&contextID=" + escapeParam(contextId);
+      log.log(Level.FINEST, "Downloading content from {0}",
+          urlString);
+      sendUrlResponse(response, urlString);
+      break;
+     }
+    }
+  }
+
+  private void sendUrlResponse(Response response, String urlString)
```

Comment:
This code is all pretty localized, and there's no javadoc here. Is this method bearing enough weight?